### PR TITLE
Add configuration error output to ots job

### PR DESCRIPTION
**Pull request by oyvindeide**

### CODE REVIEW

File: semeio/jobs/scripts/overburden_timeshift.py [hunk 1]
```diff
@@ -38,14 +39,13 @@ def _get_args_parser():
     return parser
 
 
-def main_entry_point(args=None):
+def main_entry_point():
```
Comment (lars-petter-hauge) [inline semeio/jobs/scripts/overburden_timeshift.py:42]: If we want to test the `main_entry_point` without modifying `sys.args` at runtime (and without using `subprocess`), we want to allow the `main_entry_point` to accept arguments I believe.

File: semeio/jobs/scripts/overburden_timeshift.py [hunk 2]
```diff
@@ -38,14 +39,13 @@ def _get_args_parser():
...
     parser = _get_args_parser()
-    options = parser.parse_args(args)
+    options = parser.parse_args()
     if options.help_params:
         print(generate_rst_doc())
     else:
-        ots_run(options.config)
-
-
-if __name__ == "__main__":
```
Comment (lars-petter-hauge) [inline semeio/jobs/scripts/overburden_timeshift.py:50]: Removing the `if __name__...` I would agree on!

File: semeio/jobs/scripts/overburden_timeshift.py (hunk 1 above)
Comment (oyvindeide) [reply semeio/jobs/scripts/overburden_timeshift.py:41]: Dont want to adapt production code to adhere to tests, can always mock `sys.args`?

File: semeio/jobs/overburden_timeshift/ots.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 from semeio.jobs.overburden_timeshift.ots_vel_surface import OTSVelSurface
 from semeio.jobs.overburden_timeshift.ots_res_surface import OTSResSurface
 from semeio.jobs.overburden_timeshift.ots_config import build_schema
+from semeio._exceptions.exceptions import ConfigurationError
```
Comment (xjules) [inline semeio/jobs/overburden_timeshift/ots.py:17]: The other jobs should use it too I reckon

File: tests/jobs/overburden_timeshift/test_ots_config.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        },
+    }
+    with tmpdir.as_cwd():
+        with open("ots_config.yml", "w") as f:
+            yaml.dump(conf, f, default_flow_style=False)
+        ots_load_params("ots_config.yml")
+
+
+def test_invalid_config(tmpdir, monkeypatch):
+    dates = []
```
Comment (xjules) [inline tests/jobs/overburden_timeshift/test_ots_config.py:41]: context_mock can go away and it should trigger the ConfigurationError anyway

File: semeio/jobs/scripts/overburden_timeshift.py (hunk 1 above)
Comment (oyvindeide) [reply semeio/jobs/scripts/overburden_timeshift.py:41]:
```
import argparse
import sys
def test_argparse(monkeypatch):
    monkeypatch.setattr(sys, "argv", ["path", "-c", "config_file"])
    parser = argparse.ArgumentParser()
    parser.add_argument(
        "-c", "--config"
    )
    options = parser.parse_args()
```

File: tests/jobs/overburden_timeshift/test_ots_config.py (hunk 4 above)
Comment (oyvindeide) [reply tests/jobs/overburden_timeshift/test_ots_config.py:41]: It also triggers `util_abort`

File: semeio/jobs/scripts/overburden_timeshift.py (hunk 1 above)
Comment (lars-petter-hauge) [reply semeio/jobs/scripts/overburden_timeshift.py:41]: Alright, that's a reasonable way of doing it, I think. And of course we're only mocking specifically sysargv, and not anything else

File: tests/jobs/overburden_timeshift/test_ots_config.py (hunk 4 above)
Comment (lars-petter-hauge) [reply tests/jobs/overburden_timeshift/test_ots_config.py:41]: Then there might be another issue? Why is util abort triggered? (He asked, well knowing the answer can be awkward ;) )

Comment (oyvindeide) [reply tests/jobs/overburden_timeshift/test_ots_config.py:41]: There is another issue, related to reading an ecl file which isnt ecl I think. My plan is to create an issue once this is merged (so I can reference the code). The configuration should in general have better test coverage.

File: semeio/jobs/overburden_timeshift/ots.py (hunk 3 above)
Comment (oyvindeide) [reply semeio/jobs/overburden_timeshift/ots.py:17]: If you agree that this is the right approach, then yes I agree, we could do this for the rest of the scripts as well.

File: tests/jobs/overburden_timeshift/test_ots_config.py (hunk 4 above)
Comment (lars-petter-hauge) [reply tests/jobs/overburden_timeshift/test_ots_config.py:41]: That shouldn't throw a util abort, but a more general error which can be caught (although that error is thrown for a lot of things..). Util abort is rather thrown if the ecl file is corrupt, and ecl can't recover.. but for later I guess

**Review by lars-petter-hauge — APPROVED** [review]
Alright, lgtm

File: semeio/jobs/scripts/overburden_timeshift.py (hunk 2 above)
Comment (oyvindeide) [reply semeio/jobs/scripts/overburden_timeshift.py:50]: I will move the console scripts to another PR

**Pull request merged**